src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -690,6 +690,10 @@ private function getFiltersParameters(bool $v3, string $resourceClass, string $o
                 $type = \in_array($data['type'], Type::$builtinTypes, true) ? $this->jsonSchemaTypeFactory->getType(new Type($data['type'], false, null, $data['is_collection'] ?? false)) : ['type' => 'string'];
                 $v3 ? $parameter['schema'] = $type : $parameter += $type;
 
+                if ($v3 && isset($data['schema'])) {
```

Comment:
Shouldn't we do this test before calling `$this->jsonSchemaTypeFactory->getType`, in order to not call it when it's useless (when `data['schema']` exists)? Also, are the provided schema incompatible with Swagger v2?

Reply:
We already do useless work like this in many other places in this class. (At least it was the case before. Not sure after your refactoring.)

I guess it's okay in this case as it keeps the code as simple as possible.

> Also, are the provided schema incompatible with Swagger v2?

There was no support for schema in parameters.

Reply:
> We already do useless work like this in many other places in this class. (At least it was the case before. Not sure after your refactoring.)

I tried to improve it as much as possible :) 

> I guess it's okay in this case as it keeps the code as simple as possible.

It doesn't increase the complexity that much:

```php
                if ($v3 && isset($data['schema'])) {
                    $parameter['schema'] = $data['schema'];
                } else {
                 $type = \in_array($data['type'], Type::$builtinTypes, true) ? $this->jsonSchemaTypeFactory->getType(new Type($data['type'], false, null, $data['is_collection'] ?? false)) : ['type' => 'string'];
                $v3 ? $parameter['schema'] = $type : $parameter += $type;
              }
```

(it could probably be more simplified btw)

>It has no support for schema in parameters.

It is supported: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#fixed-fields-7 (`If in is "body":`).

Reply:
And because it is supported, the code can be simplified a lot:

```php
$type = $data['schema'] ?? \in_array($data['type'], Type::$builtinTypes, true) ? $this->jsonSchemaTypeFactory->getType(new Type($data['type'], false, null, $data['is_collection'] ?? false)) : ['type' => 'string'];
$v3 ? $parameter['schema'] = $type : $parameter += $type;
```

Reply:
No, it's not supported. That's for the request body. We're talking about filters here.